### ISSUE TITLE
Can't build app on Ubuntu

### ISSUE THREAD

**Issue by rrthomas** [kovidgoyal/kitty#894]
Having installed from source, I missed having the icon, so I tried `make app`, but this fails because it wants `iconutil` from XCode. Could this be detected on a per-platform basis? `app` seems to be the right target, there's code in there for building XDG `.desktop` files…

(As a workaround I installed an earlier version as a .deb.)

**Comment by rrthomas** [comment]
Thanks. I didn't read these instructions before because I'm not trying to package kitty, I just want to install it from source. The packaging instructions say "do not install it in site-packages", and say that it gets installed to a staging area. Unfortunately this doesn't tell me how to simply install it from source, so that, for example, the icons get installed where the system can see them.

**Comment by kovidgoyal** [comment]
there is no install from source -- kitty is designed to run from a self contained directory. So if you want to install it simply run linux-package and copy the desltop and icon files into ~/.local adjusting paths as appropriate. 

A better approach would be to install your distro kitty and simply have your custom built one higher in PATH. Then the distro package desktop file should launch the custom built kitty with icon and everything.

**Comment by rrthomas** [comment]
Installing my distro kitty is what I did yesterday. But there may be many users whose distro has not (yet?) packaged kitty. And the recipe you give for running from the build directory is unsatisfactory precisely because many systems won't pick up the icon, and possibly other system-wide resources (I spot a `.desktop` file for example). (A self-contained build presumably works on macOS, and I'm a RISC OS user of old, so I know some systems don't insist on installing application files in special directories!)

Would you accept a patch to add an `install` target to the `Makefile`? Compared to the `linux-package` directory contents, the user and file permissions need fixing.

**Comment by kovidgoyal** [comment]
No, I really dont want to maintain an install target. Installing software on linux is a pain -- there are a gazillion things that can go wrong with it and I dont want to be responsible for debugging them.

**Comment by rrthomas** [comment]
You don't have to make it your problem (you can simply politely say you can't debug such problems but welcome patches.) The trouble is that at present, users who don't use one of the packaging systems you package for (not just GNU/Linux users, but *BSD etc.) have either to package the program themselves for their system (which is a fair amount of work), or run kitty from a self-contained directory (which doesn't work well on FHS-based and similar systems). Given that one of kitty's goals is to be highly portable, this is a shame.

**Comment by kovidgoyal** [comment]
Sorry, if I implement a feature in my software, it becomes my problem. At least that's the goal I strive for, I have to admit I dont always attain it.

And as far as I know running applications in self contained directories works perfectly fine on all OSes. THe only problem is that some desktop environments wont have launchers/icons for the app. I'm afraid that's not a serious enough problem for me to take on the responsibility of creating an install from source function.
